### PR TITLE
All orders are created, never destroyed.

### PR DESCRIPTION
**Pull request by karbassi**
If a user wants to cancel, the order, the 'active' flag is set to False.
If they want to sign up again, a new order is created.

### CODE REVIEW

**Comment by karbassi** [comment]
@bkellgren can you review the code changes here and give your thoughts.

File: coderdojochi/views.py [hunk 1]
```diff
@@ -400,8 +400,9 @@ def session_sign_up(request, year, month, day, slug, session_id, student_id=Fals
                 session_obj.mentors.add(mentor)
         else:
             if user_signed_up:
-                order = get_object_or_404(Order, student=student, session=session_obj)
-                order.delete()
+                order = get_object_or_404(Order, student=student, session=session_obj, active=True)
+                order.active = False
+                order.save()
```
Comment (karbassi) [inline coderdojochi/views.py]: @bkellgren Biggest change is in this file.

File: coderdojochi/views.py [hunk 2]
```diff
@@ -400,8 +400,9 @@ def session_sign_up(request, year, month, day, slug, session_id, student_id=Fals
                 session_obj.mentors.add(mentor)
         else:
             if user_signed_up:
-                order = get_object_or_404(Order, student=student, session=session_obj)
-                order.delete()
+                order = get_object_or_404(Order, student=student, session=session_obj, active=True)
```
Comment (bkellgren) [inline coderdojochi/views.py]: I think for this query you can leave out active=True because whether or not it was set properly to active=True, you want to change it to False.  Just in case for some reason active is unset, they will still be marked inactive rather than a 404. So it would be this...

` order = get_object_or_404(Order, student=student, session=session_obj)`

File: coderdojochi/views.py [hunk 3]
```diff
@@ -409,7 +410,7 @@ def session_sign_up(request, year, month, day, slug, session_id, student_id=Fals
                 else:
                     ip = request.META['REMOTE_ADDR']
 
-                order, created = Order.objects.get_or_create(guardian=guardian, student=student, session=session_obj, ip=ip)
+                order = Order.objects.create(guardian=guardian, student=student, session=session_obj, ip=ip)
```
Comment (bkellgren) [inline coderdojochi/views.py]: Why did you remove the ability to update the row in case it was already created?

Comment (karbassi) [inline coderdojochi/views.py]: Because we don't want to UPDATE the inactive ticket.

The cycle should be:

```
Sign up -> Cancel; 
Resignup = new Ticket
```

File: coderdojochi/views.py (hunk 2 above)
Comment (karbassi) [inline coderdojochi/views.py]: @bkellgren That's what I thought but it wasn't working when I tried it. Let me retest.

File: coderdojochi/views.py (hunk 3 above)
Comment (bkellgren) [inline coderdojochi/views.py]: Okay, that makes sense.
